spyder/api/widgets/__init__.py

```diff
@@ -79,6 +81,22 @@ class PluginMainContainer(QWidget, SpyderWidgetMixin, SpyderToolbarMixin):
     subclass this.
     """
 
+    CONTEXT_NAME = None
+    """
+    This attribute defines the context name under which actions, toolbars,
+    toolbuttons and menus should be registered on Spyder global registry.
+
+    If actions, toolbars, toolbuttons or menus belong to the global scope of
+    the plugin, then this attribute should have a `None` value.
+
+    Parameters
+    ----------
+    CONTEXT_NAME: Optional[str]
+        Name of the context that is used to store this container actions,
+        toolbars, toolbuttons and menus. If None, then it defaults to the
+        global plugin context.
```

Comment:
```suggestion
```

I think this is not necessary because this is a class constant and not a callable. I suggest to change this part with

> This is an optional attribute and it needs to be a string.